pkg/backoff/backoff_test.go

```diff
@@ -0,0 +1,67 @@
+package backoff
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestImmediate(t *testing.T) {
+	t.Parallel()
+
+	bkoff := New()
+	bkoff.delay = 0.0
+
+	err := bkoff.Run(willSucceed)
+	require.NoError(t, err)
+	require.Equal(t, 1, bkoff.count)
+}
+
+func TestEventual(t *testing.T) {
+	t.Parallel()
+	bkoff := New()
+	bkoff.delay = 0.0
+
+	tTime := &takesTime{
+		attempts: 0,
+	}
+
+	err := bkoff.Run(tTime.eventual)
+	require.NoError(t, err)
+	require.Equal(t, 5, bkoff.count)
+	require.Equal(t, 5, tTime.attempts)
+
+}
+
+func TestSlowFail(t *testing.T) {
+	t.Parallel()
+	bkoff := New()
+	bkoff.delay = 0.0
+
+	err := bkoff.Run(willFail)
+	require.Error(t, err)
+	require.Equal(t, 21, bkoff.count)
```

Comment:
since we _do_ have the tests, should we just fix the fencepost error?

Reply:
Okay